src/core/algorithms/algebraic_constraints/ac_algorithm.cpp

```diff
@@ -256,40 +256,50 @@ void ACAlgorithm::PrintRanges(std::vector<model::TypedColumnData> const& data) c
     }
 }
 
-std::vector<std::byte const*> ACAlgorithm::ConstructDisjunctiveRanges(ACPairs const& ac_pairs,
-                                                                      double weight) const {
+std::vector<std::byte const*> ACAlgorithm::ConstructDisjunctiveRanges(
+        ACPairs const& ac_pairs) const {
     std::vector<std::byte const*> ranges;
     if (ac_pairs.size() < 2) {
         ranges = std::vector<std::byte const*>();
         return ranges;
     }
+
     ACPair const* l_border = ac_pairs.front().get();
     ACPair const* r_border = nullptr;
-    double delta = num_type_->Dist(ac_pairs.front()->GetRes(), ac_pairs.back()->GetRes()) *
-                   (weight / (1 - weight));
-
-    for (size_t i = 0; i < ac_pairs.size() - 1; ++i) {
-        if (num_type_->Dist(ac_pairs[i]->GetRes(), ac_pairs[i + 1]->GetRes()) <= delta) {
-            r_border = ac_pairs[i + 1].get();
-        } else {
-            ranges.emplace_back(l_border->GetRes());
-            ranges.emplace_back(ac_pairs[i]->GetRes());
-            l_border = ac_pairs[i + 1].get();
-            r_border = ac_pairs[i + 1].get();
+
+    if (weight_ < 1) {
+        double delta = num_type_->Dist(ac_pairs.front()->GetRes(), ac_pairs.back()->GetRes()) *
+                       (weight_ / (1 - weight_));
+
+        for (size_t i = 0; i < ac_pairs.size() - 1; ++i) {
+            if (num_type_->Dist(ac_pairs[i]->GetRes(), ac_pairs[i + 1]->GetRes()) <= delta) {
+                r_border = ac_pairs[i + 1].get();
+            } else {
+                ranges.emplace_back(l_border->GetRes());
+                ranges.emplace_back(ac_pairs[i]->GetRes());
+                l_border = ac_pairs[i + 1].get();
+                r_border = ac_pairs[i + 1].get();
+            }
         }
+    } else {
+        assert(weight_ == 1);
+        r_border = ac_pairs.back().get();
     }
+
     if (r_border == ac_pairs.back().get()) {
         ranges.emplace_back(l_border->GetRes());
         ranges.emplace_back(r_border->GetRes());
     }
+
     return ranges;
 }
 
 RangesCollection ACAlgorithm::ReconstructRangesByColumns(size_t lhs_i, size_t rhs_i,
-                                                         double weight) const {
+                                                         double weight) {
     ACPairsCollection const& constraints_collection = GetACPairsByColumns(lhs_i, rhs_i);
     ACPairs const& ac_pairs = constraints_collection.ac_pairs;
-    std::vector<std::byte const*> ranges = ConstructDisjunctiveRanges(ac_pairs, weight);
+    SetOption(config::names::kWeight, weight);
```

Comment:
nit: I think this line should be at the beginning of the method so it fails earlier if a wrong value is passed and so that whoever reads the code sees that this sets an option.

Reply:
Agree, good point